app/util/util.ts

```diff
@@ -1,18 +1,23 @@
-import { MIST_PER_SUI } from "@mysten/sui/utils";
 import * as bitcoin from "bitcoinjs-lib";
 import Wallet, { Address } from "sats-connect";
 import { fetchUTXOs, fetchValidateAddress } from "~/api/api";
 import { nBTC_ADDR } from "~/constants";
 import { ToastFunction } from "~/hooks/use-toast";
 import { UTXO, ValidateAddressI } from "~/types";
 import { Transaction } from "@mysten/sui/transactions";
+import BigNumber from "bignumber.js";
+import { SATOSHIS_PER_BTC, MIST_PER_SUI } from "~/lib/denoms";
 
-const mistToSui = (amountInMist: number): number => {
-	return amountInMist / Number(MIST_PER_SUI);
+const mistToSui = (amountInMist: BigNumber): BigNumber => {
```

Comment:
Are you sure it's not a rounded number? 
Example 12333...  should equal 1,2333.. SUI.

Reply:
no, it's not getting auto rounded.